scripts/data/src/org/akvo/gae/remoteapi/CountFormInstances.java

```diff
@@ -0,0 +1,183 @@
+/*
+ *  Copyright (C) 2017,2019 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.gae.remoteapi;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.repackaged.org.apache.commons.io.FileUtils;
+import java.io.File;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/*
+ */
+public class CountFormInstances implements Process {
+
+  private Map<Long, String> sgName = new HashMap<>();
+  private Map<Long, String> sgType = new HashMap<>();
+  private Map<Long, Long> sgParents = new HashMap<>();
+
+  private Map<Long, String> surveyNames = new HashMap<>();
+  private Map<Long, String> surveyDates = new HashMap<>();
+  private Map<Long, Long> surveyParents = new HashMap<>();
+  private Map<Long, Long> surveyCounts = new HashMap<>();
+  private Map<Long, List<Long>> surveyToGroups = new HashMap<>();
+
+  private String iName = "";
+
+  @Override
+  public void execute(DatastoreService ds, String[] args) throws Exception {
+
+    if (args.length != 1) {
+      throw new IllegalArgumentException("Missing instance name");
+    }
+
+    iName = args[0];
```

Comment:
Does this work? I kind of recall that you would need to pass the instance id twice at the commandline if you want it to appear here. Otherwise its stripped off in the `RemoteAPI.main()` method.

Reply:
yes, because this script is only work with my bash script, please check: https://github.com/akvo/akvo-flow/blob/0cf577c2f8be72dec6c3283dac45677e2b426e5d/scripts/data/count-forminstances.sh#L21-L22

Why do we need iName? because we need to have instance_name for the first column of CSV file